src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -141,6 +142,30 @@ private Task OnProjectFactoryCompletedAsync()
             }
         }
 
+        private JoinableTask<T> ExecuteWithinLockAsync<T>(Func<Task<T>> task)
+        {
+            return JoinableFactory.RunAsync(async delegate
+            {
+                using (JoinableCollection.Join())
```

Comment:
@lifengl This type derives from OnceInitializedOnceDisposedAsync, which has these protected properties, see here: https://github.com/dotnet/roslyn-project-system/blob/637bfdf74ccf9ab2de20556e43cd54d48904a691/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs#L18